### PR TITLE
Precommit using validate_arguments

### PR DESCRIPTION
**Pull request by tilk**
This PR makes the `precommit` method in ROB runnable only if the `rob_id` from its argument matches. This allows to eliminate `m.If` expressions comparing `rob_id`s in FUs.

What do you think? Is this a good idea? I kind of like it because this seems to fit better to Transactron's philosophy: actions which are impossible due to external factors are not performed because of methods not being ready.

This PR depends on #663 and #679.

TODO:
* [x] Solve #678 
* [x] Fix tests

### CODE REVIEW

File: transactron/testing/testbenchio.py [hunk 1]
```diff
@@ -99,6 +99,7 @@ def method_handle(
         enable: Optional[Callable[[], bool]] = None,
         validate_arguments: Optional[Callable[..., bool]] = None,
         extra_settle_count: int = 0,
+        disable_after: bool = True,
```
Comment (lekcyjna123) [inline transactron/testing/testbenchio.py:102]: Where this argument is used?

File: coreblocks/func_blocks/fu/lsu/dummyLsu.py [hunk 2]
```diff
@@ -299,11 +299,10 @@ def _():
 
         with Transaction().body(m):
             precommit = self.dependency_manager.get_dependency(InstructionPrecommitKey())
-            info = precommit(m)
-            with m.If(info.rob_id == request_rob_id):
-                m.d.comb += rob_id_match.eq(1)
+            info = precommit(m, request_rob_id)
+            m.d.comb += rob_id_match.eq(1)
             with m.If(~info.side_fx):
-                m.d.comb += flush.eq(1)
```
[the remark this answers is not in the document]
Comment (tilk) [reply coreblocks/func_blocks/fu/lsu/dummyLsu.py:149]: Changed to use `flushing`.

File: coreblocks/backend/retirement.py [hunk 3]
```diff
@@ -53,11 +53,14 @@ def __init__(
...
 
+        layouts = self.gen_params.get(RetirementLayouts)
         self.dependency_manager = DependencyContext.get()
-        self.core_state = Method(o=self.gen_params.get(RetirementLayouts).core_state, nonexclusive=True)
+        self.core_state = Method(o=layouts.core_state, nonexclusive=True)
         self.dependency_manager.add_dependency(CoreStateKey(), self.core_state)
 
-        self.precommit = Method(o=self.gen_params.get(RetirementLayouts).precommit, nonexclusive=True)
+        self.precommit = Method(
+            i=layouts.precommit_in, o=layouts.precommit_out, nonexclusive=True, combiner=lambda m, args, runs: args[0]
```
[the remark this answers is not in the document]
Comment (tilk) [reply coreblocks/backend/retirement.py]: Added a comment, and replaced args[0] with just 0 to make it clear that the combiner here is a dummy.

File: transactron/testing/testbenchio.py (hunk 1 above)
Comment (tilk) [reply transactron/testing/testbenchio.py]: Nowhere. Removed.

**Review by piotro888 — APPROVED** [review]

**Pull request merged**